### ISSUE TITLE
Hostname environment variable no longer works in elasticsearch.yml when starting using systemd

### ISSUE THREAD

**Issue by shaharmor** [elastic/elasticsearch#26255]
**Elasticsearch version** (`bin/elasticsearch --version`):
6 beta 1

**Plugins installed**: []

**JVM version** (`java -version`):
```
java version "1.8.0_144"
Java(TM) SE Runtime Environment (build 1.8.0_144-b01)
Java HotSpot(TM) 64-Bit Server VM (build 25.144-b01, mixed mode)
```

**OS version** (`uname -a` if on a Unix-like system):

`Linux elasticsearch-data-001 4.4.0-92-generic #115-Ubuntu SMP Thu Aug 10 09:04:33 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux`

**Description of the problem including expected versus actual behavior**:

Before ES 6-beta1, usage of `${HOSTNAME}` in the `elasticsearch.yml` config file was possible for the `node.name` config parameter.
It no longer works in ES 6-beta1, probably following this PR: https://github.com/elastic/elasticsearch/pull/25807

**Steps to reproduce**:

Please include a *minimal* but *complete* recreation of the problem, including
(e.g.) index creation, mappings, settings, query etc.  The easier you make for
us to reproduce it, the more likely that somebody will take the time to look at it.

 1. Install ES 6-beta1 on Ubuntu 16.04
 2. Start using systemd: `systemctl restart elasticsearch.service`
 3. Observe logs using: `journalctl -u elasticsearch.service`

**Provide logs (if relevant)**:

Relevant info copied from PR:
This is my elasticsearch.yml file:
```yml
cluster.name: test
path.data: /data/elasticsearch
path.logs: /var/log/elasticsearch
discovery.zen.ping.unicast.hosts:
  - 'elasticsearch-data-001'
  - 'elasticsearch-data-002'
  - 'elasticsearch-data-003'

node.name: ${HOSTNAME}
node.master: true
node.data: true
node.max_local_storage_nodes: 1

network.host:
  - '_site_'
  - '_local_'

bootstrap.memory_lock: true
http.cors.enabled: true
http.cors.allow-origin: "*"
```

OS:
```
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.3 LTS
Release:	16.04
Codename:	xenial
```

And `HOSTNAME` is an existing environment variable:

```
root@elasticsearch-data-001:~# bash -c 'echo $HOSTNAME'
elasticsearch-data-001
```

When starting using `systemctl restart elasticsearch.service`, i'm getting the following error in `journalctl -u elasticsearch.service`:
```
Aug 17 11:22:38 elasticsearch-data-001 systemd[1]: Started Elasticsearch.
Aug 17 11:22:50 elasticsearch-data-001 elasticsearch[104026]: Exception in thread "main" java.lang.IllegalArgumentException: Could not resolve placeholder 'HOSTNAME'
Aug 17 11:22:50 elasticsearch-data-001 elasticsearch[104026]:         at org.elasticsearch.common.settings.PropertyPlaceholder.parseStringValue(PropertyPlaceholder.java:116)
Aug 17 11:22:50 elasticsearch-data-001 elasticsearch[104026]:         at org.elasticsearch.common.settings.PropertyPlaceholder.replacePlaceholders(PropertyPlaceholder.java:69)
Aug 17 11:22:50 elasticsearch-data-001 elasticsearch[104026]:         at org.elasticsearch.common.settings.Settings$Builder.replacePropertyPlaceholders(Settings.java:1117)
Aug 17 11:22:51 elasticsearch-data-001 systemd[1]: elasticsearch.service: Main process exited, code=exited, status=1/FAILURE
Aug 17 11:22:51 elasticsearch-data-001 systemd[1]: elasticsearch.service: Unit entered failed state.
Aug 17 11:22:51 elasticsearch-data-001 systemd[1]: elasticsearch.service: Failed with result 'exit-code'.
```
This is after an apt-get dist-upgrade that upgraded ES from latest alpha to beta1. In the alpha it worked flawlessly.

@jasontedor

**Comment by colings86** [comment]
@jasontedor are you able to take a look at this?

**Comment by jasontedor** [comment]
Thanks for reporting @shaharmor. This would have been a terrible inadvertent breaking change to release 6.0.0 with. We really appreciate you trying out the beta and added you to the [Pioneer Program](https://www.elastic.co/blog/elastic-pioneer-program-6-0). I reproduced the issue and opened #26262. This time I added a test so this does not get inadvertently broken again. This will be fixed in the next pre-release version of Elasticsearch 6.0.0.
